### PR TITLE
[Feature-8023][Document] Add example and notice about task type Python

### PR DESCRIPTION
**Pull request by EricGao888**
- Based on the [shell task example](https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/task/shell.html), this pr adds example and notice about task type Python
- This pr closes: apache/dolphinscheduler#8023

### CODE REVIEW

**Comment by EricGao888** [comment]
> * Based on the [shell task example](https://dolphinscheduler.apache.org/en-us/docs/latest/user_doc/guide/task/shell.html), this pr adds example and notice about task type Python
> * This pr closes: #8023

@zhongjiajie  FYi, this pr is WIP. I only fix the python task examples in English dev version in this commit. If things look good, I will fix python task examples in other versions of docs in the following commits. Thx : )

**Comment by zhongjiajie** [comment]
will task a look tomorrow, is midnight now

**Review by zhongjiajie — COMMENTED** [review]
Looking good! thanks @EricGao888. FYI you just need to change `dev` and latest(for now is`2.0.3`) doc in both EN and ZH doc. we could jus ignore previous verisons

**Comment by EricGao888** [comment]
> Looking good! thanks @EricGao888. FYI you just need to change `dev` and latest(for now is`2.0.3`) doc in both EN and ZH doc. we could jus ignore previous verisons

Sure, thx for the comment! Maybe we can adopt jinjava or other template lib to generate doc in the feture? I think that may save a lot of work.

**Comment by zhongjiajie** [comment]
Good point, yeah, we already have a plan in https://github.com/apache/dolphinscheduler/issues/6535, section `DRY`. But our community do not have time to do it

**Comment by zhongjiajie** [comment]
BTW, we plan to move our docs into main repo. After that we only need to change dev version docs. (in most cases)

**Comment by EricGao888** [comment]
> Looking good! thanks @EricGao888. FYI you just need to change `dev` and latest(for now is`2.0.3`) doc in both EN and ZH doc. we could jus ignore previous verisons

@zhongjiajie I've changed the `dev` and latest(2.0.5) doc in both EN and ZH versions. PTAL

**Review by zhongjiajie — COMMENTED** [review]
docs looking good, but could you change the screenshot to latest in branch dev? We change our UI recently which you can see https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/task/flink.html as example

**Comment by EricGao888** [comment]
> docs looking good, but could you change the screenshot to latest in branch dev? We change our UI recently which you can see https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/task/flink.html as example

Sure, the new UI looks cool!

**Comment by EricGao888** [comment]
> docs looking good, but could you change the screenshot to latest in branch dev? We change our UI recently which you can see https://dolphinscheduler.apache.org/en-us/docs/dev/user_doc/guide/task/flink.html as example

Have changed the screenshots in dev directory to match the newest UI : )

**Comment by zhongjiajie** [comment]
I approved the CI running just now

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 1]
```diff
@@ -1,15 +1,55 @@
...
-<p align="center">
-   <img src="/img/python-en.png" width="80%" />
- </p>
+## Create Task
 
-- Script: Python program developed by the user.
-- Environment Name: Specific Python interpreter path for running the script. If you need to use Python **virtualenv**, you should create multiply environments for each **virtualenv**.  
-- Resources: Refers to the list of resource files that need to be called in the script.
-- User-defined parameter: It is a user-defined local parameter of Python, and will replace the content with `${variable}` in the script.
-- Note: If you import the python file under the resource directory tree, you need to add the `__init__.py` file.
```
Comment (zhongjiajie) [inline docs/en-us/dev/user_doc/guide/task/python.md:15]: BTW, @Tianqi-Dotes could you make a double check about our dev EN document change?

Comment (zhongjiajie) [reply docs/en-us/dev/user_doc/guide/task/python.md:15]: I find that we have some docs overwrite in other PR

Comment (EricGao888) [reply docs/en-us/dev/user_doc/guide/task/python.md:15]: @zhongjiajie @Tianqi-Dotes Is there anything else I could do here? I mean the docs overwrite stuff, not sure what it is.

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 2]
```diff
@@ -1,15 +1,55 @@
 # Python Node
 
-- Using python nodes, you can directly execute python scripts. For python nodes, workers use `python **` to submit tasks.
+## Overview
 
-> Drag from the toolbar ![PNG](https://analysys.github.io/easyscheduler_docs_cn/images/toolbar_PYTHON.png) task node to the canvas, as shown in the following figure:
+`Python Task` is used to create a python-type task and execute python scripts. When the worker execute `Python Task`,
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:5]: better
`Python Task` is used to create
change to
Use `Python Task` to create

Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:5]: When the worker execute
change to 
When the worker executes

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 3]
```diff
@@ -1,15 +1,55 @@
...
+a temporary python script is generated, and the Linux user with the same name as the tenant executes the script.
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:6]: better
a temporary python script is generated,
change to 
will generate a temporary python script,

Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:6]: ', and the Linux user with the same name as the tenant executes the script.' 
should be 
', and executes the script by the Linux user with the same name as the tenant .'

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 4]
```diff
@@ -1,15 +1,55 @@
...
- </p>
+## Create Task
 
-- Script: Python program developed by the user.
-- Environment Name: Specific Python interpreter path for running the script. If you need to use Python **virtualenv**, you should create multiply environments for each **virtualenv**.  
-- Resources: Refers to the list of resource files that need to be called in the script.
-- User-defined parameter: It is a user-defined local parameter of Python, and will replace the content with `${variable}` in the script.
-- Note: If you import the python file under the resource directory tree, you need to add the `__init__.py` file.
+- Click Project Management-Project Name-Workflow Definition, and click the "Create Workflow" button to enter the DAG editing page.
+- Drag <img src="/img/tasks/icons/python.png" width="15"/> from the toolbar to the drawing board.
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:11]: drawing board
to 
canvas

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 5]
```diff
@@ -1,15 +1,55 @@
...
+
+- Node name: The node name in a workflow definition is unique.
+- Run flag: Identifies whether this node can be scheduled normally, if it does not need to be executed, you can turn on the prohibition switch.
+- Descriptive information: describe the function of the node.
+- Task priority: When the number of worker threads is insufficient, they are executed in order from high to low, and when the priority is the same, they are executed according to the first-in first-out principle.
+- Worker grouping: Tasks are assigned to the machines of the worker group to execute. If Default is selected, a worker machine will be randomly selected for execution.
+- Environment Name: Configure the environment name in which to run the script.
+- Number of failed retry attempts: The number of times the task failed to be resubmitted. It supports drop-down and hand-filling.
+- Failed retry interval: The time interval for resubmitting the task after a failed task. It supports drop-down and hand-filling.
+- Timeout alarm: Check the timeout alarm and timeout failure. When the task exceeds the "timeout period", an alarm email will be sent and the task execution will fail.
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:23]: will be sent
to
will send

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 6]
```diff
@@ -1,15 +1,55 @@
...
+- Script: PYTHON program developed by users.
+- Resource: Refers to the list of resource files that need to be called in the script, and the files uploaded or created by the resource center-file management.
+- Custom parameters: It is a user-defined parameter that is part of PYTHON, which will replace the content with \${variable} in the script.
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:26]: It is a user-defined parameter that is part of PYTHON
to
It is the user-defined parameters of Python,

File: docs/en-us/dev/user_doc/guide/task/python.md [hunk 7]
```diff
@@ -1,15 +1,55 @@
...
+
+```python
+print("This is a demo of python task")
+```
+
+### Custom Parameters
+
+This example is a sample custom parameter task which could reuse existing as template, or for dynamic task. First of all,
+we should declare a custom parameter named "param_key", with the value as "param_val". Then we using keyword "${param_key}"
+to using the parameter we just declared. After this example is being run, we would see "param_val" print in the log
```
Comment (Tianqi-Dotes) [inline docs/en-us/dev/user_doc/guide/task/python.md:45]: better change to
This example simulates a custom parameter task. We use parameters for reusing existing tasks as template or coping with the dynamic task. In this case,
we declare a custom parameter named "param_key", with the value "param_val". Then we use `echo` to print the parameter "${param_key}" we just declared. 
After running this example, we would see "param_val" print in the log.

**Review by Tianqi-Dotes — COMMENTED** [review]
Hi Mr. Gao,
please try not to use passive patterns in technology writing. Thanks for the contribution.
Yours

**Comment by zhongjiajie** [comment]
Thanks @Tianqi-Dotes  for reviewing

File: docs/en-us/dev/user_doc/guide/task/python.md (hunk 2 above)
Comment (EricGao888) [reply docs/en-us/dev/user_doc/guide/task/python.md]: Fixed in the latest commit.

**Comment by EricGao888** [comment]
> Hi Mr. Gao, please try not to use passive patterns in technology writing. Thanks for the contribution. Yours

@Tianqi-Dotes Hi Tianqi, thx for helping me review the docs in such detail. I've fixed the related language issues and resolved the comments : )

**Comment by EricGao888** [comment]
> I will merge it when it finishes the CI, and then will do the doc migration

@zhongjiajie Cool, I'm looking forward to the doc migration. After migration, it seems easier for developers to contribute to dolphin docs and I'd like to keep contributing to these docs : )

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks and I will migrate it ASAP

**Pull request merged**